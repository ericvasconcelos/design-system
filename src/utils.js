export { arrayCombinations } from 'utils/arrayCombinations'
export { getBackgroundColor } from 'utils/getBackgroundColor'
export { getBorder } from 'utils/getBorder'
export { getButtonStyle } from 'utils/getButtonStyle'
export { getBorderRadius } from 'utils/getBorderRadius'
export { getColorFromName } from 'utils/getColorFromName'
export { getGuttersNames } from 'utils/getGuttersNames'
export { getIconNames } from 'utils/getIconNames'
export { getPadding } from 'utils/getPadding'
export { getPaddingNames } from 'utils/getPaddingNames'
export { getTypography } from 'utils/getTypography'
export { parsePalleteName } from 'utils/parsePalleteName'
export { styleColorValidator } from 'utils/styleColorValidator'
export { useAriaProps } from 'utils/useAriaProps'

export { DateUtils } from 'utils/DateUtils'
export { QueryUtils } from 'utils/QueryUtils'
export { ObjectUtils }  from 'utils/ObjectUtils'
export { CookiesUtils } from 'utils/CookiesUtils'
export { MoneyUtils } from 'utils/MoneyUtils'
export { NumberUtils } from 'utils/NumberUtils'