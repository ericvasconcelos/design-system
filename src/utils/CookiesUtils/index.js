export class CookiesUtils {
  static toObject(string = '') {
    const cookies = {}
    if (string !== '') {
      const parts = string.split('; ')
      parts.forEach(part => {
        const [key, value] = part.split('=')
        cookies[key] = value
      })
    }
    return cookies
  }
}
