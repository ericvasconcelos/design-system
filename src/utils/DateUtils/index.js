export class DateUtils {
  static toDate(string, format = 'pt-br') {
    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = string?.split('/')
        break
      case 'iso':
        [year, month, day] = string?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    return new Date(year, parseInt(month, 10) - 1, day)
  }

  static getAge(date) {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const difference = Date.now() - date.getTime()

    return Math.abs(new Date(difference).getUTCFullYear() - 1970)
  }

  static toString(date, format = 'pt-br') {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
    const month = date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
    const year = date.getFullYear()

    switch (format) {
      case 'pt-br':
        return `${day}/${month}/${year}`
      case 'iso':
        return `${year}-${month}-${day}`
      default:
        throw new Error('Format invalid')
    }
  }

  static validate(date, format = 'pt-br') {
    const monthDaysMap = {
      '01': 31,
      '02': 29,
      '03': 31,
      '04': 30,
      '05': 31,
      '06': 30,
      '07': 31,
      '08': 31,
      '09': 30,
      10: 31,
      11: 30,
      12: 31
    }
    let isValid = false

    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = date?.split('/')
        break
      case 'iso':
        [year, month, day] = date?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    isValid = isValid || parseInt(day, 10) <= monthDaysMap[month]
    isValid = isValid && parseInt(year, 10) > 1900

    return isValid
  }

  static convertMonthToYear(month) {
    return parseInt(month, 10) / 12
  }

  static formatPeriod(period, periodType) {
    const periodTypeMap = {
      monthly: { singular: 'mÃªs', plural: 'meses' },
      yearly: { singular: 'ano', plural: 'anos' },
      daily: { singular: 'dia', plural: 'dias' }
    }
    return `${period} ${periodTypeMap[periodType][period < 2 ? 'singular' : 'plural']}`
  }
}
