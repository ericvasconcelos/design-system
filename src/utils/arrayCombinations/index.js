export const arrayCombinations = (...args) => {
  const returnable = []
  const max = args.length - 1
  function helper(arr, i) {
    for (let j = 0, { length } = args[i]; j < length; j++) {
      const newArray = arr.slice(0) // clone arr
      newArray.push(args[i][j])
      if (i === max) returnable.push(newArray)
      else helper(newArray, i + 1)
    }
  }
  helper([], 0)
  return returnable
}
