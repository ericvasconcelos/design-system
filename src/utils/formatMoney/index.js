import { toMoney, toNumber } from 'vanilla-masker'

export const formatMoney = entry => {
  const moneyFormatter = {
    precision: 2,
    separator: ',',
    delimiter: '.',
    unit: 'R$'
  }

  return toMoney(toNumber(entry), moneyFormatter)
}
