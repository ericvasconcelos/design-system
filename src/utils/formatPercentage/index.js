import { toMoney, toNumber } from 'vanilla-masker'

export const formatPercentage = entry => {
  const percentageFormatter = {
    precision: 1,
    separator: ',',
    suffixUnit: '%'
  }
  return toMoney(toNumber(entry), percentageFormatter)
}
