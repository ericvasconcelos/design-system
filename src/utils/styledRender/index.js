import React from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from 'styles/theme'

export const styledRender = (children, options = {}) => {
  const rendered = render(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  return {
    ...rendered,
    rerender: (ui, rerenderOptions = {}) =>
      styledRender(ui, { container: rendered.container, ...rerenderOptions })
  }
}
