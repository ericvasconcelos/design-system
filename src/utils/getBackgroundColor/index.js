import { getColorFromName } from 'utils/getColorFromName'

export const getBackgroundColor = ({ theme, styleColor }) =>
  `background-color: ${getColorFromName({ theme, styleColor })};`
