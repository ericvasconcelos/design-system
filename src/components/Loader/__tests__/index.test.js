import React from 'react'

import { styledRender } from 'utils/styledRender'

import { Loader } from '../index.js'

test('should render the component correctly', () => {
  const { getByTestId } = styledRender(<Loader />)

  const component = getByTestId('Loader')

  expect(component).toBeInTheDocument()
  expect(component).toMatchSnapshot()
})

test('should render the component corectly when has children', () => {
  const { getByTestId, getByText } = styledRender(
    <Loader>
      <p>message</p>
    </Loader>
  )

  const component = getByTestId('Loader')

  expect(component).toBeInTheDocument()
  expect(getByText('message')).toBeInTheDocument()
  expect(component).toMatchSnapshot()
})
