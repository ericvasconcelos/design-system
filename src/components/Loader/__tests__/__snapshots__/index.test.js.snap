// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component corectly when has children 1`] = `
.c0 {
  margin-top: -6em;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 0.625rem;
  margin: 0 auto;
  text-indent: -9999em;
  width: 6em;
  height: 6em;
  border-color: white;
  border-radius: 50%;
  background: #003FF2;
  background: -moz-linear-gradient( left, #003FF2 10%, white 42% );
  background: -webkit-linear-gradient( left, #003FF2 10%, white 42% );
  background: -o-linear-gradient( left, #003FF2 10%, white 42% );
  background: -ms-linear-gradient( left, #003FF2 10%, white 42% );
  background: linear-gradient( to right, #003FF2 10%, white 42% );
  position: relative;
  -webkit-animation: load3 1.4s infinite linear;
  -webkit-animation: load3 1.4s infinite linear;
  animation: load3 1.4s infinite linear;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c1:before {
  width: 50%;
  height: 50%;
  background: #003FF2;
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}

.c1:after {
  background: #FFFFFF;
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

<div
  class="c0"
  data-testid="Loader"
>
  <div
    class="c1"
  />
  <p>
    message
  </p>
</div>
`;

exports[`should render the component correctly 1`] = `
.c0 {
  margin-top: -6em;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 0.625rem;
  margin: 0 auto;
  text-indent: -9999em;
  width: 6em;
  height: 6em;
  border-color: white;
  border-radius: 50%;
  background: #003FF2;
  background: -moz-linear-gradient( left, #003FF2 10%, white 42% );
  background: -webkit-linear-gradient( left, #003FF2 10%, white 42% );
  background: -o-linear-gradient( left, #003FF2 10%, white 42% );
  background: -ms-linear-gradient( left, #003FF2 10%, white 42% );
  background: linear-gradient( to right, #003FF2 10%, white 42% );
  position: relative;
  -webkit-animation: load3 1.4s infinite linear;
  -webkit-animation: load3 1.4s infinite linear;
  animation: load3 1.4s infinite linear;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c1:before {
  width: 50%;
  height: 50%;
  background: #003FF2;
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}

.c1:after {
  background: #FFFFFF;
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

<div
  class="c0"
  data-testid="Loader"
>
  <div
    class="c1"
  />
</div>
`;
